server/src/test/java/e2e/UserTaskTest.java

```diff
@@ -71,6 +77,22 @@ void shouldTransferOwnershipFromUserToSpecificGroupOnDeadline() {
         Assertions.assertThat(wfRunIdList.getResultsList()).isNotEmpty();
     }
 
+    @Test
+    void shouldTestDeadlineReassignmentWithEpoch() {
+        WfRunVerifier wfRunVerifier = workflowVerifier
+                .prepareRun(deadlineReassignmentWithEpoch)
+                .waitForStatus(RUNNING)
+                .waitForUserTaskRunStatus(1, 1, UserTaskRunStatus.ASSIGNED)
+                .waitForNodeRunStatus(2, 1, COMPLETED, Duration.ofSeconds(8))
+                .thenAssignUserTask(1, 1, true, "anakin", null)
+                .waitForUserTaskRunStatus(1, 1, UserTaskRunStatus.UNASSIGNED, Duration.ofSeconds(20));
```

Comment:
Don't think we need to wait `20` seconds here, see my comment below.

Reply:
Test becomes flaky if I add a short sleep time here. I'm not sure if this test is worth it.

Reply:
Hmm. Yeah, we can delete this test *for now* but we should open a ticket to figure out a better way to implement it.